logic/wonko/WonkoVersion.h

```diff
@@ -0,0 +1,105 @@
+/* Copyright 2015 MultiMC Contributors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include "BaseVersion.h"
+#include "BaseWonkoEntity.h"
+
+#include <QVector>
+#include <QStringList>
+#include <QJsonObject>
+#include <memory>
+
+#include "WonkoReference.h"
+
+#include "multimc_logic_export.h"
+
+using WonkoVersionPtr = std::shared_ptr<class WonkoVersion>;
+
+class MULTIMC_LOGIC_EXPORT WonkoVersion : public QObject, public BaseVersion, public BaseWonkoEntity
```

Comment:
This is a weird mixture of things that do not belong in once class. Qt-style self-aware object, basic Wonko structural data and Minecraft specific data.

I disagree with this being all jumbled together.